src/svxlink/modules/metarinfo/ModuleMetarInfo.cpp

```diff
@@ -661,30 +655,61 @@ void ModuleMetarInfo::allMsgsWritten(void)
 
 } /* allMsgsWritten */
 
+/*
+* callback function, called-up by libcurl framework
+*/
+size_t ModuleMetarInfo::callback(char *contents, size_t size, size_t nmemb,
+                                       void *userp)
+{
+  if (userp == NULL) return 0;
+  size_t written = size * nmemb;
+  char *t = (char *)malloc(written +1);
+  memcpy(t, (const char *)contents, written);
+  t[written] = '\0';
+  static_cast<ModuleMetarInfo *>(userp)->onDataReceived((std::string)t, 
+                                                   (std::string(t)).length());
+  free(t);
+  return written;
+} /* MetarInfo::callback */
+
 
 /*
-* establish a tcp-connection to the METAR-Server
+* establish a https-connection to the METAR-Server
+* using curl library
 */
-void ModuleMetarInfo::openConnection(void)
+void ModuleMetarInfo::openConnection()
 {
-  if (con == 0)
-  {
-    con = new TcpClient<>(server, 80);
-    con->connected.connect(mem_fun(*this, &ModuleMetarInfo::onConnected));
-    con->disconnected.connect(mem_fun(*this, &ModuleMetarInfo::onDisconnected));
-    con->dataReceived.connect(mem_fun(*this, &ModuleMetarInfo::onDataReceived));
-    con->connect();
+  CURL *curl;
+  CURLcode res;
+  
+  curl_global_init(CURL_GLOBAL_DEFAULT);
+  curl = curl_easy_init();
+ 
+  std::string path = server;
+              path += link;
+              path += icao;
+
+  if(curl) {
+    curl_easy_setopt(curl, CURLOPT_URL, path.c_str());
+    curl_easy_setopt(curl, CURLOPT_VERBOSE, 0L);
+    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, &ModuleMetarInfo::callback);
+    curl_easy_setopt(curl, CURLOPT_WRITEDATA, this);
+    res = curl_easy_perform(curl);
```

Comment:
A quick check in the documentation say that this call blocks until done. That is very bad in SvxLink since that will freeze all operations until the whole web page have been fetched. I think you should look at [curl_multi_perform](https://curl.haxx.se/libcurl/c/curl_multi_perform.html) instead or use some other strategy that does not have blocking behavior.

Reply:
You will need to get the file descriptor for the https connection and then set up a Async::FdWatch to receive the data.